public/app/features/scopes/selector/ScopesSelectorService.ts

```diff
@@ -142,7 +144,9 @@ export class ScopesSelectorService extends ScopesServiceBase<ScopesSelectorServi
     if (selectedIdx === -1) {
       // We prefetch the scope when clicking on it. This will mean that once the selection is applied in closeAndApply()
       // we already have all the scopes in cache and don't need to fetch all of them again is multiple requests.
-      this.apiClient.fetchScope(linkId!);
+      this.apiClient.fetchScope(linkId!).then((scope) => {
+        setRecentScopes(scope);
```

Comment:
Setting the recent scopes when the actual scopes are being selected is preferred, but we need to do a comparison between new and existing. And how do we handle if the selected scopes are more than 5 (current arbitrary limit)